OrbitVulkanLayer/DispatchTable.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitVulkanLayer/DispatchTable.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit::layer {
+
+void DispatchTable::CreateInstanceDispatchTable(const VkInstance& instance,
+                                                const PFN_vkGetInstanceProcAddr& next_gipa) {
+  // Create dispatch table
+  VkLayerInstanceDispatchTable dispatch_table = {};
+  dispatch_table.DestroyInstance =
+      absl::bit_cast<PFN_vkDestroyInstance>(next_gipa(instance, "vkDestroyInstance"));
+  dispatch_table.GetInstanceProcAddr =
+      absl::bit_cast<PFN_vkGetInstanceProcAddr>(next_gipa(instance, "vkGetInstanceProcAddr"));
+  dispatch_table.EnumerateDeviceExtensionProperties =
+      absl::bit_cast<PFN_vkEnumerateDeviceExtensionProperties>(
+          next_gipa(instance, "vkEnumerateDeviceExtensionProperties"));
+  dispatch_table.EnumeratePhysicalDevices = absl::bit_cast<PFN_vkEnumeratePhysicalDevices>(
+      next_gipa(instance, "vkEnumeratePhysicalDevices"));
+
+  // Store the dispatch table for this instance
+  {
+    absl::WriterMutexLock lock(&mutex_);
+    instance_dispatch_table_[instance] = dispatch_table;
+  }
+}
+
+void DispatchTable::RemoveInstanceDispatchTable(const VkInstance& instance) {
+  absl::WriterMutexLock lock(&mutex_);
+  instance_dispatch_table_.erase(instance);
+}
+
+void DispatchTable::CreateDeviceDispatchTable(const VkDevice& device,
+                                              const PFN_vkGetDeviceProcAddr& next_gdpa) {
+  // Create dispatch table
```

Comment:
Remove comment. This is redundant as the method name already tells you what the code is doing.